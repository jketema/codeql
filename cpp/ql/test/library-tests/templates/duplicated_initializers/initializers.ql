import cpp

from Variable v
where v.getName() = "is_same_v"
select v,
  concat(int index, string res | res = v.getTemplateArgument(index).toString() + "[" + index + "]" | res, ","  order by index),
  count(v.getInitializer().getExpr())
