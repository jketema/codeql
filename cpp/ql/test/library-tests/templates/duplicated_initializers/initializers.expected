| template_var_initializers.cpp:1:40:1:40 | is_same_v | _Ty1[0],_Ty[0],int[0],long[0],_Ty2[1],_Ty[1],int[1],long[1] | 5 |
| template_var_initializers.cpp:1:40:1:40 | is_same_v | _Ty[0],_Ty[1] | 0 |
| template_var_initializers.cpp:1:50:1:50 | is_same_v | _Ty1[0],_Ty[0],int[0],long[0],_Ty2[1],_Ty[1],int[1],long[1] | 5 |
| template_var_initializers.cpp:2:27:2:27 | is_same_v | _Ty1[0],_Ty[0],int[0],long[0],_Ty2[1],_Ty[1],int[1],long[1] | 5 |
| template_var_initializers.cpp:2:27:2:45 | is_same_v | _Ty1[0],_Ty[0],int[0],long[0],_Ty2[1],_Ty[1],int[1],long[1] | 5 |
